Replace the TestMailServer to GreenMail server

rachana-dani opened the pull request:
### Description
[Describe what this change achieves]
Switched TestMailServer to GreenMail server

### Issues Resolved
[List any issues this PR will resolve]
This is to stop using deprecated test smtp server

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Hailong-am commented:
@rachana-dani can you fix the DCO check? you need add `-s` flag with git commit command

Hailong-am commented on notifications/core/src/test/kotlin/org/opensearch/notifications/core/smtp/SmtpEmailTests.kt:
```diff
@@ -14,28 +17,29 @@ import org.opensearch.notifications.core.transport.SmtpDestinationTransport
+
+    @BeforeEach
+    fun setUpServer() {
+        greenMail = GreenMail(ServerSetupTest.SMTP)
```
what's the default port does test server used?

rachana-dani replied:
This is typically 3025

zhichao-aws reviewed (commented):
Is there a method to verify whether the GreenMail server recieve the notification?

rachana-dani commented:
> Is there a method to verify whether the GreenMail server recieve the notification?

There is a method greenMail.receivedMessages. I am checking size of the returned object

rachana-dani commented:
> > Is there a method to verify whether the GreenMail server recieve the notification?
> 
> There is a method greenMail.receivedMessages. I am checking size of the returned object

I have addressed all the comments. Please take a look.

rachana-dani commented:
> > > Is there a method to verify whether the GreenMail server recieve the notification?
> > 
> > 
> > There is a method greenMail.receivedMessages. I am checking size of the returned object
> 
> I have addressed all the comments. Please take a look.

@Hailong-am @zhichao-aws all the checks have passed.

Hailong-am commented on notifications/core/build.gradle:
```diff
@@ -161,6 +161,10 @@ dependencies {
     )
     testImplementation 'org.springframework.integration:spring-integration-mail:5.5.0'
+    // https://mvnrepository.com/artifact/com.icegreen/greenmail
+    testImplementation group: 'com.icegreen', name: 'greenmail', version: '1.5.0'
```
is it possible we use latest version of greenmail? I do see there have two CVEs with version 1.5.0 https://mvnrepository.com/artifact/com.icegreen/greenmail/1.5.0

rachana-dani commented:
Can you please add a HackoberFest2023 label and help me merge this PR?

Hailong-am reviewed (approved):
Looks good to me

Hailong-am commented on notifications/core/build.gradle:
```diff
@@ -161,6 +161,8 @@ dependencies {
     )
     testImplementation 'org.springframework.integration:spring-integration-mail:5.5.0'
+    // https://mvnrepository.com/artifact/com.icegreen/greenmail
+    testImplementation group: 'com.icegreen', name: 'greenmail', version: '1.6.4'
```
```suggestion
    testImplementation group: 'com.icegreen', name: 'greenmail', version: '1.6.14'
```

rachana-dani replied:
I have updated this version. Pls R&A

SuZhou-Joe commented:
#789
